src/battle_anim_new.c

```diff
@@ -9151,11 +9151,14 @@ void AnimTask_GetWeatherToSet(u8 taskId)
 
 void AnimTask_SyrupBomb(u8 taskId)
 {
-    struct Pokemon *party = GetBattlerParty(gBattleAnimAttacker);
-    u32 isShiny = IsMonShiny(&party[gBattlerPartyIndexes[gBattleAnimAttacker]]);
+    if (!(gStatuses4[gBattleAnimTarget] & STATUS4_SYRUP_BOMB))
```

Comment:
Again, this must be tested in link battles because `gStatuses4` is normally only set by the host GBA
Also, did you mean to do `if (gStatuses4[gBattleAnimTarget] & STATUS4_SYRUP_BOMB)`?

Reply:
>  Again, this must be tested in link battles because gStatuses4 is normally only set by the host GBA
Mentioned in the Syrup Bomb PR ([screenshot](https://user-images.githubusercontent.com/93446519/282052157-59b06afd-4985-4fd3-b150-17fbb8b1abd2.png)) but it seemed fine using link battles on mGBA. But also good chance I'm misunderstand what I should be looking for. Though I'm also not sure what I would use instead of `gStatus4`  

> Also, did you mean to do if (gStatuses4[gBattleAnimTarget] & STATUS4_SYRUP_BOMB)?
Done some more testing and I messed up here. I'll see if I can find a proper solution.

Reply:
So the idea was to check `STATUS4_SYRUP_BOMB` and set the shiny field if the status wasn't applied yet. Though applying it to the whole animation block messed it up. I ended up doing the same but only to the shiny field. It does seem to work.

https://github.com/rh-hideout/pokeemerald-expansion/assets/93446519/85b33b1c-0916-4566-85b1-289b05d3e175